libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipParams.kt

```diff
@@ -231,7 +231,13 @@ data class GossipParams(
      * callback to notify outer system to which peers Gossip wants to be connected
      * The second parameter is a signed peer record: https://github.com/libp2p/specs/pull/217
      */
-    val connectCallback: (PeerId, ByteArray) -> Unit = { _: PeerId, _: ByteArray -> }
+    val connectCallback: (PeerId, ByteArray) -> Unit = { _: PeerId, _: ByteArray -> },
+
+    /**
+     * [maxIDontWantMessages] is the maximum number of IDONTWANT messages per heartbeat per peer
+     */
+    val maxIDontWantMessages: Int = 10
```

Comment:
I believe it's pretty low default which may mitigate the whole purpose of `IDONTWANT`. 
I would default it to `maxIHaveLength * maxIHaveMessages` as proposed [here](https://github.com/libp2p/specs/pull/548#discussion_r1523551738)

This one would be pretty high but any client may make it more conservative if needed.   

With respect to Ethereum PeerDAS we may legally have up to 128 messages in a single slot for data column subnets